lms/djangoapps/grades/tasks.py

```diff
@@ -115,6 +115,25 @@ def recalculate_subsection_grade_v3(self, **kwargs):
     _recalculate_subsection_grade(self, **kwargs)
 
 
+import time
+from celery import group
+@task()
+def add(x, y):
+    time.sleep(4)
+    return x + y
+
+
+@task
+def tsum(x):
+    time.sleep(8)
+    return sum(x)
+
+
+@task
+def do_stuff():
+    thing = group(add.s(i, i) for i in xrange(10))
+    thing.apply_async()
```

Comment:
@nasthagiri, I can now confirm that the main task will not need to hang around when using groups:

```
edxapp@vagrant:~/edx-platform$ ./manage.py lms --settings=devstack_with_worker shell
Python 2.7.12 (default, Nov 19 2016, 06:48:10) 
[GCC 5.4.0 20160609] on linux2
Type "help", "copyright", "credits" or "license" for more information.
(InteractiveConsole)
>>> from lms.djangoapps.grades.tasks import do_stuff
>>> do_stuff.apply_async()
<AsyncResult: 120e7255-6618-43b2-9bdf-684a6a3ae2e4>
```

```
edxapp@vagrant:~/edx-platform$ ./manage.py lms celery worker --settings=devstack_with_worker -l DEBUG

<truncated>

2017-04-20 00:40:08,500 WARNING 2297 [celery.redirected] log.py:282 - celery@vagrant ready.
2017-04-20 00:40:29,034 INFO 2297 [celery.worker.strategy] strategy.py:55 - Received task: lms.djangoapps.grades.tasks.do_stuff[120e7255-6618-43b2-9bdf-684a6a3ae2e4]
2017-04-20 00:40:29,051 INFO 2297 [celery.worker.strategy] strategy.py:55 - Received task: lms.djangoapps.grades.tasks.add[bd7ab137-f998-4629-9ae6-80863d6c649f]
2017-04-20 00:40:29,055 INFO 2297 [celery.worker.strategy] strategy.py:55 - Received task: lms.djangoapps.grades.tasks.add[51e4b47d-f5c1-4be2-ba59-ddbd9f1ee3e0]
2017-04-20 00:40:29,055 INFO 2297 [celery.worker.strategy] strategy.py:55 - Received task: lms.djangoapps.grades.tasks.add[80fafb84-c0b0-454d-ba77-4adfad2b805c]
2017-04-20 00:40:29,061 INFO 2297 [celery.worker.job] job.py:401 - **Task lms.djangoapps.grades.tasks.do_stuff[120e7255-6618-43b2-9bdf-684a6a3ae2e4] succeeded in 0.0251774170029s**: None
2017-04-20 00:40:29,062 INFO 2297 [celery.worker.strategy] strategy.py:55 - Received task: lms.djangoapps.grades.tasks.add[e1f3b5cd-65cc-4a41-9a0d-54b02a2b4119]
2017-04-20 00:40:33,062 INFO 2297 [celery.worker.job] job.py:401 - Task lms.djangoapps.grades.tasks.add[bd7ab137-f998-4629-9ae6-80863d6c649f] succeeded in 4.01016565299s: 0
2017-04-20 00:40:33,064 INFO 2297 [celery.worker.strategy] strategy.py:55 - Received task: lms.djangoapps.grades.tasks.add[ee218ff7-1de4-415b-a210-ea171857fffd]
2017-04-20 00:40:33,067 INFO 2297 [celery.worker.job] job.py:401 - Task lms.djangoapps.grades.tasks.add[51e4b47d-f5c1-4be2-ba59-ddbd9f1ee3e0] succeeded in 4.00562678199s: 2
2017-04-20 00:40:33,069 INFO 2297 [celery.worker.strategy] strategy.py:55 - Received task: lms.djangoapps.grades.tasks.add[4deb70e1-62a4-4d14-8cbc-e1d95af191dc]
2017-04-20 00:40:37,068 INFO 2297 [celery.worker.job] job.py:401 - Task lms.djangoapps.grades.tasks.add[80fafb84-c0b0-454d-ba77-4adfad2b805c] succeeded in 4.00515334899s: 4
2017-04-20 00:40:37,070 INFO 2297 [celery.worker.strategy] strategy.py:55 - Received task: lms.djangoapps.grades.tasks.add[6e12717c-aaa6-43f2-80a1-66b25067445e]
2017-04-20 00:40:37,072 INFO 2297 [celery.worker.job] job.py:401 - Task lms.djangoapps.grades.tasks.add[e1f3b5cd-65cc-4a41-9a0d-54b02a2b4119] succeeded in 4.00382452298s: 6
2017-04-20 00:40:37,073 INFO 2297 [celery.worker.strategy] strategy.py:55 - Received task: lms.djangoapps.grades.tasks.add[814a8c85-d376-4370-b3f4-a487d0c8eeb9]
2017-04-20 00:40:41,076 INFO 2297 [celery.worker.job] job.py:401 - Task lms.djangoapps.grades.tasks.add[ee218ff7-1de4-415b-a210-ea171857fffd] succeeded in 4.00636468502s: 8
2017-04-20 00:40:41,076 INFO 2297 [celery.worker.job] job.py:401 - Task lms.djangoapps.grades.tasks.add[4deb70e1-62a4-4d14-8cbc-e1d95af191dc] succeeded in 4.00390823698s: 10
2017-04-20 00:40:41,078 INFO 2297 [celery.worker.strategy] strategy.py:55 - Received task: lms.djangoapps.grades.tasks.add[8bd6f764-b378-45e9-b44f-d7c30b45c772]
2017-04-20 00:40:41,079 INFO 2297 [celery.worker.strategy] strategy.py:55 - Received task: lms.djangoapps.grades.tasks.add[2c96d1e1-067d-4bb9-abd5-34d0c0e52703]
2017-04-20 00:40:45,083 INFO 2297 [celery.worker.job] job.py:401 - Task lms.djangoapps.grades.tasks.add[6e12717c-aaa6-43f2-80a1-66b25067445e] succeeded in 4.00615953599s: 12
2017-04-20 00:40:45,084 INFO 2297 [celery.worker.job] job.py:401 - Task lms.djangoapps.grades.tasks.add[814a8c85-d376-4370-b3f4-a487d0c8eeb9] succeeded in 4.007225913s: 14
2017-04-20 00:40:49,091 INFO 2297 [celery.worker.job] job.py:401 - Task lms.djangoapps.grades.tasks.add[2c96d1e1-067d-4bb9-abd5-34d0c0e52703] succeeded in 4.00612286798s: 18
2017-04-20 00:40:49,091 INFO 2297 [celery.worker.job] job.py:401 - Task lms.djangoapps.grades.tasks.add[8bd6f764-b378-45e9-b44f-d7c30b45c772] succeeded in 4.006555019s: 16
```